GUIDELINES.md

```diff
@@ -0,0 +1,153 @@
+* Don't compare a value to `None` (`value == None`), always favour `value is None`
+
+### Encouraged
+
+* Favour `logging` over `print`
+* Favour using `GLOBAL_VARIABLES` near the top of your code rather than *any* hard-coded variables in your code. These can be more easily factored out into config, and at least puts many of your design decisions in one place.
+
+
+## Critical thinking
+
+* Surely this simple problem already has an elegant solution?
+* How many copies of the data am I making?
+* Where do my variables go out of scope?
+* How can I avoid creating a new variable at all costs? (think iterator, scope, `lru_cache`)
+
+## Naming conventions
+
+### Mandatory
+
+* Functions / methods: `function`, `my_function`
+* Variables / attributes: `variable`, `my_var`
+* Class: `Model`, `MyClass`
+* Module / file names / directory names: `module`, `file_name.py`, `dir_name`
+* Global* / constants: `A_GLOBAL_VARIABLE`
+
+\* Global at the module level, not the `global` level. Don't use `global`, ever.
+
+### Encouraged
+
+* Keep all names as short and descriptive as possible. Variable names such as `x` or `df` are highly discouraged unless they are genuinely representing abstract concepts.
+* Favour good naming conventions over helpful comments
+
+
+## Spaces and spacing
+
+### Encouraged
+
+* Use the absolute minimum number of indents of your code. You can achieve this by writing modular code, and inverting logic, for example:
+
+```python
+def something(args):
+	for item_collection in args:          # 1 tab
+		if item_collection.exists():      # 2 tabs
+			the_sum = 0                   # 3 tabs
+			for item in item_collection:
+				the_sum += item.value()   # 4 tabs
+			print(the_sum)
+```
+
+can become:
+
+```python
+def sum_values(item_collection):
+	the_sum = sum(item.value() for item in item_collection)
+	print(the_sum)
+
+def something(args):
+	for item_collection in args:          # 1 tab
+		if not item_collection.exists():  # 2 tabs
+			continue
+		sum_values(item_collection)
+```
+
+* Put a space before starting block comments `# like this`, `#not this`
+* Inline comments need two spaces before them `a = 2  # like this`
+* Keep lines to 79 characters or less. You can achieve this by utilising other parts of this guideline, particularly with regards to creating modular code. Splitting over multiple lines is, of course, permissible so long as it doesn't conflict with legibility.
+* When declaring default values, never put spaces around operators like `=`, i.e `def this_is_ok(param=1)`, `def this_is_NOT_ok(param = 1)`. Otherwise, all operators must always have a single space on either side.
+
+
+## Comments and docs
+
+### Mandatory
+
+* At least *a basic docstring* is required for every function/method and class
+* Full, explanatory docstrings are required for all function/methods and classes *if* it will be used in the main body of a code routine.
+
+### Encouraged
+
+* Don't state the obvious in comments
+
```

Comment:
* Before writing a comment, consider whether that information would be better be encoded in a variable or function name.

Reply:
(:point_up:  this was also a suggestion)